tests/test_packages/test_skills/test_echo.py

```diff
@@ -20,160 +20,67 @@
+    write_envelope,
+)
+from aea.test_tools.test_cases import AEATestCase
 
-from aea.cli import cli
-from aea.configurations.base import AgentConfig, DEFAULT_AEA_CONFIG_FILE, PublicId
-from aea.mail.base import Envelope
-from aea.protocols.default.message import DefaultMessage
-from aea.protocols.default.serialization import DefaultSerializer
 
-from ...common.click_testing import CliRunner
-from ...conftest import AUTHOR, CLI_LOG_OPTION
-
-
-class TestEchoSkill:
+class TestEchoSkill(AEATestCase):
     """Test that echo skill works."""
 
-    @classmethod
-    def setup_class(cls):
-        """Set up the test class."""
-        cls.runner = CliRunner()
-        cls.agent_name = "my_first_agent"
-        cls.cwd = os.getcwd()
-        cls.t = tempfile.mkdtemp()
-        os.chdir(cls.t)
-
     def test_echo(self, pytestconfig):
         """Run the echo skill sequence."""
         if pytestconfig.getoption("ci"):
             pytest.skip("Skipping the test since it doesn't work in CI.")
 
-        # add packages folder
-        packages_src = os.path.join(self.cwd, "packages")
-        packages_dst = os.path.join(self.t, "packages")
-        shutil.copytree(packages_src, packages_dst)
+        self.initialize_aea()
+        agent_name = "my_first_agent"
+        self.create_agents(agent_name)
 
-        result = self.runner.invoke(
-            cli,
-            [*CLI_LOG_OPTION, "init", "--local", "--author", AUTHOR],
-            standalone_mode=False,
-        )
-        assert result.exit_code == 0
-
-        # create agent
-        result = self.runner.invoke(
-            cli,
-            [*CLI_LOG_OPTION, "create", "--local", self.agent_name],
-            standalone_mode=False,
-        )
-        assert result.exit_code == 0
-        agent_dir_path = os.path.join(self.t, self.agent_name)
+        agent_dir_path = os.path.join(self.t, agent_name)
         os.chdir(agent_dir_path)
 
-        # disable logging
-        aea_config_path = Path(self.t, self.agent_name, DEFAULT_AEA_CONFIG_FILE)
-        aea_config = AgentConfig.from_json(yaml.safe_load(open(aea_config_path)))
-        aea_config.logging_config = {
-            "disable_existing_loggers": False,
-            "version": 1,
-            "loggers": {"aea.echo_skill": {"level": "CRITICAL"}},
-        }
-        yaml.safe_dump(dict(aea_config.json), open(aea_config_path, "w"))
-
-        # add skills
-        result = self.runner.invoke(
-            cli,
-            [*CLI_LOG_OPTION, "add", "--local", "skill", "fetchai/echo:0.1.0"],
-            standalone_mode=False,
+        self.add_item("skill", "fetchai/echo:0.1.0")
+
+        process = self.run_agent()
+        time.sleep(2.0)
+
+        # add sending and receiving envelope from input/output files
+        message = create_default_message(b"hello")
+        expected_envelope = create_envelope(agent_name, message)
+
+        write_envelope(expected_envelope)
+
+        time.sleep(2.0)
+        lines = readlines_output_file()
```

Comment:
I think the lines 65-75 should be wrapped in a method called `read_envelope_from_file`, mirroring the above